cmd/minikube/cmd/delete.go

```diff
@@ -210,23 +210,28 @@ func DeleteProfiles(profiles []*config.Profile) []error {
 	klog.Infof("DeleteProfiles")
 	var errs []error
 	for _, profile := range profiles {
-		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
-		defer cancel()
-		err := deleteProfile(ctx, profile)
-		if err != nil {
-			mm, loadErr := machine.LoadMachine(profile.Name)
+		errs = append(errs, deleteProfileTimeout(profile)...)
+	}
+	return errs
+}
 
-			if !profile.IsValid() || (loadErr != nil || !mm.IsValid()) {
-				invalidProfileDeletionErrs := deleteInvalidProfile(profile)
-				if len(invalidProfileDeletionErrs) > 0 {
-					errs = append(errs, invalidProfileDeletionErrs...)
-				}
-			} else {
-				errs = append(errs, err)
+func deleteProfileTimeout(profile *config.Profile) []error {
```

Comment:
thanks for breaking down this long code to more modular one, this is more readable !

Reply:
there was a small issue with `defer cancel()` actually. When called inside the `for` loop, cancel() is deferred till the end of the function, not the loop iteration